azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -391,9 +401,12 @@ def _fetch_full_wire_server_goal_state(self, incarnation, xml_doc):
                 certs = Certificates(xml_text, self.logger)
                 # Log and save the certificates summary (i.e. the thumbprint but not the certificate itself) to the goal state history
                 for c in certs.summary:
-                    self.logger.info("Downloaded certificate {0}".format(c))
+                    message = "Downloaded certificate {0}".format(c)
+                    self.logger.info(message)
+                    add_event(op=WALAEventOperation.GoalState, message=message)
                 if len(certs.warnings) > 0:
                     self.logger.warn(certs.warnings)
+                    add_event(op=WALAEventOperation.GoalState, message=certs.warnings)
                 self._history.save_certificates(json.dumps(certs.summary))
 
```

Comment:
I left that one out, but I can add it if you think it'd be useful. If the operation fails we would get telemetry for the ProtocolError, otherwise we can assume it completed successfully.

Reply:
I'd add to avoid assumption. Especially it's useful for on callers or somebody who queries our data without knowing our code.

Reply:
In general, the telemetry skips successful operations (when possible) and reports only errors, but I'll go ahead and add this one

Reply:
added

Reply:
I agree. Since we are adding related information, I thought this would be useful to avoid confusion.  Thanks for the change.